### ISSUE TITLE
[Question] Incorrect peer dependency

### ISSUE THREAD

**Issue by kchirayus** [yarnpkg/yarn#6533]
**What is the current behavior?**
I'm getting warning incorrect peer dependency when the right version is included in the package.json

**If the current behavior is a bug, please provide the steps to reproduce.**
Ran yarn install and this is the warning i get:

`react-addons-test-utils@15.6.2" has incorrect peer dependency "react-dom@^15.4.2".`

![screen shot 2018-10-11 at 4 34 13 pm](https://user-images.githubusercontent.com/13051486/46839683-f333ee00-cd73-11e8-81a0-a29594e21915.png)

**What is the expected behavior?**
No warning

**Please mention your node.js, yarn and operating system version.**
MacOS 10.13.6
yarn 1.10.1
node 8.11.3

**Comment by arcanis** [comment]
No, the version you're using (^16.3.3) doesn't satisfy the peer dependency request (^15.4.2). The caret only allow the minor and patch versions to change, not the major.

**Comment by kchirayus** [comment]
Ah, got it. Thanks.
